.travis.yml

```diff
@@ -61,6 +61,10 @@ before_install:
   - if [[ $TRAVIS_OS_NAME == 'osx' ]]; then brew update; brew install python; brew link --overwrite python ; fi
   - pip install -U --user pip
   - pip install -U --user virtualenv
+  # Fail fast on lint errors
+  - pip install -U --user flake8
+  - if [[ $TRAVIS_OS_NAME == 'osx' ]]; then export PATH=$PATH:$HOME/Library/Python/2.7/bin; fi
+  - make lint
```

Comment:
Yes, the yml commands are executed in the repository with the to-be-tested  commit already checked out.  Hence the dance inside `firedrake-install` when running on travis.